neverwinter/nwscript/native/scriptcompparsetree.cpp

```diff
@@ -524,6 +524,59 @@ int32_t CScriptCompiler::GenerateParseTree()
 					ModifySRStackReturnTree(pNewNode);
 					return 0;
 				}
+                else if (m_nTokenStatus == CSCRIPTCOMPILER_TOKEN_KEYWORD_DASHDASH_FUNCTION)
+                {
+                    CScriptParseTreeNode *pNewNode = CreateScriptParseTreeNode(CSCRIPTCOMPILER_OPERATION_CONSTANT_STRING,NULL,NULL);
+                    pNewNode->m_psStringData = new CExoString(m_sCurrentFunction.CStr());
+                    pNewNode->m_bAllowAsDefaultValueInFunctionDecl = false;
```

Comment:
There's a `pNode->nType` which is supposed to hold `CSCRIPTCOMPILER_TOKEN_*` constants, but isn't always filled. You can just assign `m_nTokenStatus` to it and check that instead of carrying a new variable. Compressing the node data will help with graphviz and stuff.

Reply:
I'm not feeling confident I understand the ramifications here. nType seems used all over the place and overloading the variable with a different meaning just adds cruft.

Can this maybe be a follow up change that turns it into a union, if struct size needs to be kept down?